rust/src/lib/route.rs

```diff
@@ -612,7 +611,11 @@ impl MergedRoutes {
 
 // Validating if the route destination network is valid,
 // 0.0.0.0/8 and its subnet cannot be used as the route destination network
-fn validate_route_dst(dst: &str) -> Result<(), NmstateError> {
+// except for the route type is unreachable, blackhole or prohibit
+fn validate_route_dst(
+    dst: &str,
+    is_no_nexthop_route_type: bool,
```

Comment:
Please do not use `negative` for boolean variable, it will be hard for people(as least for mortal like me) to understand `is_no_nexthop_route_type: false`.

Meanwhile, instead of spreading decision making code in two place, please just pass `route_type: Option<RouteType>` to `validate_route_dst()`.

Reply:
You may do

```rust
impl RouteType {
  pub(crate) fn is_unicast(&self) -> bool;
}
```

So you may state that only unicast route to 0.0.0.0/8 is not allowed.